components/encryption/src/encrypted_file/mod.rs

```diff
@@ -70,7 +70,7 @@ impl<'a> EncryptedFile<'a> {
             .create(true)
             .write(true)
             .open(&tmp_path)
-            .unwrap();
+            .unwrap_or_else(|_| panic!("EncryptedFile::write {}", &tmp_path.to_str().unwrap()));
```

Comment:
Panic here might be simpler than propagating errors, this error is fatal, callers may panic anyway.

Reply:
It was already performing a panic. I need to limit the scope of my improvements in this large PR. There are unwraps everywhere in our code base.

Reply:
well, i actually prefer propagate the error to upper layer, which prints the error message of encryption failure, before panicking. I hope we can reduce the unnecessary panics one by one, but since the panic exists previously, it is okay not to update it.